File: Grace/JavaScript/Grace.js
/*
    Author: mhufflep
*/

const write = `require('fs').writeFile("Grace_kid.js", data(data.toString()), function(err) {});`;
const execute = (x) => { eval(x) };
const data = (x) => `/*\n    Author: mhufflep\n*/\n\nconst write = \`${write}\`;\nconst execute = ${execute.toString()};\nconst data = ${x};\nexecute(write);`;
execute(write);

File: Grace/C/Grace.c
/* Author: mhufflep */
#include <stdio.h>
#define T(x) #x
#define S(x) T(x)
#define MAIN(x) int main() { const char *s = "/* Author: mhufflep */\n#include <stdio.h>\n#define T(x) #x\n#define S(x) T(x)\n#define MAIN(x) "x"\nMAIN(S(MAIN(x)))"; FILE *f = fopen("Grace_kid.c", "w+"); fputs(s, f); fclose(f); return 0; }
MAIN(S(MAIN(x)))